src/type.cpp

```diff
@@ -1555,9 +1556,8 @@ StructType::StructType(const std::string &n, const llvm::SmallVector<const Type
         // Create a unique anonymous struct name if we have an anonymous
         // struct (name == ""), or if we are creating a derived type from
```

Comment:
1. comment needs to be update.
2. do we still create "derived type from an anonymous struct"?

Reply:
Yeah. The comment/derived part is still valid.  When variability is specified, this path is taken.

Reply:
How does this works, as it's refer to "e.g. the varying variant--name == '$'" part, which was removed?

Reply:
'$' was a kind of hack to identify later that we are dealing with anonymous struct. I never found  a case where that code was hit. If this was intended for creating separate names for anonymous when variability is mentioned, then that doesn't make sense either since we mangle the name with variability later too(lmangleStructName). The only place where we were using '$' was for printing name. I changed implementation is flag it as anonymous which I think is better.

Reply:
Ok, looks like there's no way for ISPC to generate names starting with $.

But the comment about `varying variant--name == '$'` is misleading now.